template.json

```diff
@@ -274,6 +274,7 @@
           "max": 1
         }
       },
+      "artifactBonus": "",
```

Comment:
Hmm... Either you or I have misunderstood how templates work. :) 

To my understanding each property in the `templates` object should be an object, which in turn contains the properties of that template with their default values. If you set the template to `""` I'm not sure how that will behave?

Compare with the `bonus` template above, for example.

I would suggest making the `artifactBonus` a part of the `bonus` template above. Then all objects that have the `bonus` property will automatically also have the `artifactBonus` property:

```
"templates": {
      "bonus": {
        "bonus": {
          "value": 1,
          "max": 1
        },
        "artifactBonus": ""
      },
      ...
```

If you want to keep `artifactBonus` as its own template I think you need to change it to this:

```
"templates": {
      "bonus": {
        "bonus": {
          "value": 1,
          "max": 1
        }
      },
      "artifactBonus": {
          "artifactBonus": ""
        }
      },
      ...
```

Reply:
To my knowledge this is a strange redundancy in this particular system: 
```
"templates": {
      "bonus": {
        "bonus": {}
      }
}
```
And while I have generally seen that the initial child of the templates in general be an object, this should not be required. Ref: foundryvtt.com/article/system-development/

> There is no requirement or limitation on the depth or structure of the template - however we encourage logical grouping of similarly themed attributes within named objects as this tends to help with organization and maintainability.

And as such your suggestion carries much better logical coherence than the current implementation. Also requires less space which is always good.

Reply:
So this is probably an unnecessary rant then, but I'll just put it down here in case we can learn something. I hope you don't mind... :)

The thing is, I think you're confusing the word "template" here, which is totally understandable as it kind of has double meanings within this context. The paragraph you're quoting starts with "Elements in the data model", and I believe that is what they are talking about -- the properties on the data models can indeed have any structural depth (or none). But the properties inside the "templates" sections do require objects in order to work. If you've seen otherwise, I'd be intrigued to see how!

If you want to convince yourself of this you can do what I just did. Try changing the default value of the new `artifactBonus` template, the way you added it, to be, say, `"d8"`. You would then expect new weapons to be created with a d8 artifact bonus, right? But they won't be. Unless you wrap it all in another object, as shown above, then it works as expected.

Reply:
Thank you for taking the time to educate me on this.

I did not misunderstand the meaning of template, but I did miss that the quote above was not intended to for the `template.json` so much as it was intended to describe the data model. Now, reading this part: 

> Within each entity type, the "data" entry defines shared data elements which all entities of that type have. Objects defined as siblings to "data" define specialized entity sub-types which may have data elements which extend or overwrite those defined by the basic data object.

Makes a bit more sense. Though I believe the whole section could have made a better attempt at separating description of the entity.data object and the template.json itself.

That said, you are indeed correct in that the initial child needs to be an object. If you don't mind me asking how did you come by this? Foundry APIs documentation is lacking at best, and I sorely wish to be able to understand it better.

As for the solution. I realized that the rest of the artifact properties should also be in the template. Under templates I added this entry instead:
```
      "artifact": {
        "artifactBonus": "",
        "drawback": "",
        "description": "",
        "appearance": "",
        "effect": ""
      }
```

Reply:
That sounds reasonable! :+1:

I too wish to understand Foundry's API better, so I hope we can learn together while making the Forbidden Lands experience better. :)

I'm afraid I don't have any resource to share on the subject, though. This was more a case of noticing that your template didn't follow the same pattern as the existing ones, and then thinking about it from there.

Reply:
Thanks for having this discussion. I'm still quite ignorant on this topic, and it's nice to be be able to follow along.

I do prefer the last iteration of your `artifact` template. It's much more sensible to put every artifact properties in it rather than just the artifact die.